aws-s3-transfer-manager/src/io/stream.rs

```diff
@@ -72,9 +73,19 @@ impl InputStream {
     ///     InputStream::from_path("docs/rows.csv").expect("file should be readable")
     /// }
     /// ```
-    pub fn from_path(path: impl AsRef<Path>) -> Result<InputStream, Error> {
+    pub fn from_path(path: impl AsRef<Path>) -> Result<InputStream, crate::io::error::Error> {
         Self::read_from().path(path).build()
     }
+
+    /// Converts `InputStream` to ByteStream that can be used in PutObject.
+    pub(crate) async fn into_byte_stream(self) -> Result<ByteStream, error::Error> {
+        match self.inner {
+            RawInputStream::Fs(path_body) => ByteStream::from_path(path_body.path)
+                .await
+                .map_err(error::from_kind(error::ErrorKind::InputInvalid)),
```

Comment:
**correctness**: this may not always be the right error, it could be some kind of I/O error trying to read the file metadata